datafusion/tests/sql.rs

```diff
@@ -807,17 +807,20 @@ async fn csv_query_window_with_empty_over() -> Result<()> {
     avg(c3) over (), \
     count(c3) over (), \
     max(c3) over (), \
-    min(c3) over () \
+    min(c3) over (), \
+    first_value(c3) over (), \
+    last_value(c3) over (), \
+    nth_value(c3, 2) over ()
     from aggregate_test_100 \
-    order by c2 \
+    order by c2
     limit 5";
     let actual = execute(&mut ctx, sql).await;
     let expected = vec![
-        vec!["1", "781", "7.81", "100", "125", "-117"],
-        vec!["1", "781", "7.81", "100", "125", "-117"],
-        vec!["1", "781", "7.81", "100", "125", "-117"],
-        vec!["1", "781", "7.81", "100", "125", "-117"],
-        vec!["1", "781", "7.81", "100", "125", "-117"],
+        vec!["1", "781", "7.81", "100", "125", "-117", "1", "30", "-40"],
```

Comment:
I am not sure about these results. "first_value", "last_value" and "nth_value" over an unsorted window (as this query is is using) seems undefined to me. 

For example, Using `datafusion-cli`:

```
    CREATE EXTERNAL TABLE aggregate_test_100 (
        c1  VARCHAR NOT NULL,
        c2  INT NOT NULL,
        c3  SMALLINT NOT NULL,
        c4  SMALLINT NOT NULL,
        c5  INT NOT NULL,
        c6  BIGINT NOT NULL,
        c7  SMALLINT NOT NULL,
        c8  INT NOT NULL,
        c9  BIGINT NOT NULL,
        c10 VARCHAR NOT NULL,
        c11 FLOAT NOT NULL,
        c12 DOUBLE NOT NULL,
        c13 VARCHAR NOT NULL
    )
    STORED AS CSV
    WITH HEADER ROW
    LOCATION '/Users/alamb/Software/arrow-datafusion/testing/data/csv/';
```
You can see there are many values for `c3`, and the values of `first_value`, `last_value` and `nth_value` seem to be picking some arbitrary rows: 

```
> select c2, c3 from aggregate_test_100 order by c2;
+----+------+
| c2 | c3   |
+----+------+
| 1  | 29   |
| 1  | -85  |
| 1  | 38   |
| 1  | 57   |
| 1  | 54   |
| 1  | 103  |
| 1  | -98  |
| 1  | -99  |
| 1  | -25  |
| 1  | 36   |
| 1  | 41   |
| 1  | -8   |
| 1  | -24  |
| 1  | 125  |
| 1  | 70   |
| 1  | -72  |
| 1  | 71   |
| 1  | -56  |
| 1  | -5   |
| 1  | 12   |
| 1  | 83   |
| 1  | 120  |
| 2  | 1    |
| 2  | 113  |
| 2  | 49   |
| 2  | 97   |
| 2  | -29  |
| 2  | 45   |
| 2  | -60  |
| 2  | 93   |
| 2  | 63   |
| 2  | 52   |
| 2  | 31   |
| 2  | -106 |
| 2  | -60  |
| 2  | 68   |
| 2  | -61  |
| 2  | 122  |
| 2  | -48  |
| 2  | 52   |
| 2  | -117 |
| 2  | 29   |
| 2  | -107 |
| 2  | -43  |
| 3  | 104  |
| 3  | 13   |
| 3  | 112  |
| 3  | 77   |
| 3  | 17   |
| 3  | 13   |
| 3  | 73   |
| 3  | -2   |
| 3  | 22   |
| 3  | 17   |
| 3  | -76  |
| 3  | 71   |
| 3  | 14   |
| 3  | -12  |
| 3  | -72  |
| 3  | 97   |
| 3  | -101 |
| 3  | -95  |
| 3  | 123  |
| 4  | -111 |
| 4  | -38  |
| 4  | -54  |
| 4  | -56  |
| 4  | -53  |
| 4  | 123  |
| 4  | 97   |
| 4  | 102  |
| 4  | 65   |
| 4  | 17   |
| 4  | 55   |
| 4  | 73   |
| 4  | -117 |
| 4  | -101 |
| 4  | -79  |
| 4  | 74   |
| 4  | 96   |
| 4  | -90  |
| 4  | -59  |
| 4  | 3    |
| 4  | 5    |
| 4  | 47   |
| 4  | 30   |
| 5  | -40  |
| 5  | -82  |
| 5  | 36   |
| 5  | -31  |
| 5  | -5   |
| 5  | 68   |
| 5  | -59  |
| 5  | 62   |
| 5  | -94  |
| 5  | 64   |
| 5  | -86  |
| 5  | 118  |
| 5  | -101 |
| 5  | -44  |
+----+------+
```

It is not clear that `1`, `30` and `-40` are the "right" answers (there is no good answer for this dataset)